Gems/PhysX/Code/Source/HeightfieldColliderComponent.h

```diff
@@ -34,7 +34,7 @@ namespace PhysX
     //! The heightfield collider is a bit different from the other shape colliders in that it gets the heightfield data from a
     //! HeightfieldProvider, which can control position, rotation, size, and even change its data at runtime.
     //! 
-    //! Due to these differences, this component directly implements both the collider and static rigid body services instead of
+    //! This component directly implements both the collider and static rigid body instead of
```

Comment:
Nit: IMO, the comment change removes the cause->effect linkage of the sentences, so it puts more burden on the user to infer *why* this component is avoiding the base classes. But if you think it's obvious anyways, I'm fine with that.

Reply:
Yeah, I removed that part because now it's not a necessity anymore to do it this way the Heightfield now is choosing to handle the static rigid body itself. 
Before it was a must because there was no static rigid body component, but now that there is one it could take that responsibility out and rely on the static rigid body component. This is the improvement that the **Out of Scope** section is referring to, but that's a refactor that we can do later.

Reply:
Ah, ok, revised Nit: This comment would ideally say something like "This component currently directly implements both the collider and static rigid body instead of using BaseColliderComponent and StaticRigidBodyComponent because that was previously necessary. This is no longer necessary, so this could get refactored to use those." (Ideally something a little better and less clumsy, but you get the idea)

It just seems a little weird to me to leave the comment there as a statement of fact, without explaining any reasons.

Reply:
Yeah, I agree that it would be good to add a little more context to the comment. Although having something that essentially amounts to a TODO in a doxygen comment may not be ideal.

Reply:
Let's see what I can put there in the comment to make it a more clear the current situation. `EditorStaticRigidBodyComponent` has being added, but it's still empty, for now it only forcing dependencies via services, the logic has to actually be moved there first, THEN these components can do the transition. So they are in a middle state at the moment.